Desktop/data/importers/importer.cpp

```diff
@@ -13,75 +14,127 @@ Importer::Importer()
 		DataSetPackage::pkg()->setDataSetSize(columnCount, rowCount);
-
-
-		int colNo = 0;
-		for (ImportColumn *& importColumn : *importDataSet)
+		
+		for(int colNo=0; colNo<columnCount; colNo++)
 		{
-			progressCallback(50 + 25 * colNo / columnCount);
-			initColumn(colNo, importColumn);
-			delete importColumn;
-			importColumn = nullptr;
-			colNo++;
+			ImportColumn	* importColumn	= _importDataSet->getColumn(colNo);
+			Column			* dataSetColumn	= DataSetPackage::pkg()->dataSet()->column(colNo);
+			InitColumnTask	* task			= new InitColumnTask(importColumn, dataSetColumn);
+			
+			connect(importColumn, &ImportColumn::finished, this, &Importer::importColumnFinished, Qt::DirectConnection);
+			
+			QThreadPool::globalInstance()->start(task);
 		}
-
+		
+		bool keepWaiting = true;
+		while(keepWaiting)
+		{
+			QThread::sleep(std::chrono::nanoseconds(10000));
+			_serialFinishing.lock();
+			keepWaiting = _waitingFor.size() > 0;
+			_serialFinishing.unlock();
+		}
+		
 		DataSetPackage::pkg()->dataSet()->endBatchedToDB([&](float f){ progressCallback(75 + f * 25); });
 	}
 	JASPTIMER_STOP(Importer::loadDataSet createDataSetAndLoad);
 	
-	importDataSet->clearColumns();
-	delete importDataSet;
 	DataSetPackage::pkg()->endLoadingData();
 	
-	long totalS = (Utils::currentSeconds() - timeBeginS);
-	Log::log() << "Loading '" << locator << "' took " << totalS << "s or " << (totalS / 60) << "m" << std::endl;
-}
-
-void Importer::initColumn(QVariant colId, ImportColumn *importColumn)
-{
-	JASPTIMER_SCOPE(Importer::initColumn);
-	
-	bool doLabels = !_synching || importerDeliversLabels();
+	_importDataSet->clearColumns();
+	delete _importDataSet;
 	
-	static stringvec dummyLabels;
 	
-	initColumnWithStrings(colId, importColumn->name(),  importColumn->allValuesAsStrings(), doLabels ? importColumn->allLabelsAsStrings() : dummyLabels, importColumn->title(), importColumn->getColumnType(), importColumn->allEmptyValuesAsStrings());
-}
-
-void Importer::initColumnWithStrings(QVariant colId, const std::string &newName, const std::vector<std::string> &values, const std::vector<std::string> &labels, const std::string & title, columnType desiredType, const stringset & emptyValues) 
-{ 
-	DataSetPackage::pkg()->initColumnWithStrings(colId, newName, values, labels, title, desiredType, emptyValues); 																																							 
+	long totalS = (Utils::currentSeconds() - timeBeginS);
+	Log::log() << "Loading '" << locator << "' took " << totalS << "s or " << (totalS / 60) << "m" << std::endl;
 }
 
 void Importer::syncDataSet(const std::string &locator, std::function<void(int)> progress)
 {
-	_synching = true;
-	long timeBeginS = Utils::currentSeconds();
-	
-	ImportDataSet *	importDataSet	= loadFile(locator, progress);
-	bool			rowCountChanged	= importDataSet->rowCount() != DataSetPackage::pkg()->dataRowCount();
+					_synching			= true;
+					_progressCallback	= progress;
+	long			timeBeginS		= Utils::currentSeconds();
```

Comment:
I'm worried that maximum size on MSVC and GCC is inconsistent, maybe which related to the engine timeout limit cause crash on Windows only? anyway, I recommend using `int32_t/int64_t` or something everywhere.

Reply:
I guess we could also just use the `std::chrono` datatypes now?

Interesting idea that it could be related to the timer., not sure if thats it though.

Reply:
> I guess we could also just use the `std::chrono` datatypes now?
> 

I think so.